### PR TITLE
ROX-17224: set up https metrics server

### PR DESCRIPTION
**Pull request by stehessel**
## Description

Set up a skeleton for the secure metrics settings and http server. The secure metrics endpoint is defaulted to `disabled` for now, but will eventually be enabled by default for at least Central and Sensor. See follow up PRs for the actual TLS handling.

## Checklist
- [ ] Investigated and inspected CI test results
- [x] Unit test and regression tests added
- [ ] ~Evaluated and added CHANGELOG entry if required~
- [ ] ~Determined and documented upgrade steps~
- [ ] ~Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))~

If any of these don't apply, please comment below.

### CODE REVIEW

**Comment by stehessel** [comment]
Current dependencies on/for this PR:
* master
  * **PR #6665** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/6665" target="_blank">Graphite</a> 
    * **PR #6666** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/6666" target="_blank">Graphite</a>  👈
      * **PR #6667** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/6667" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/stackrox/stackrox/6666?utm_source=stack-comment).

File: pkg/env/metrics.go [hunk 1]
```diff
@@ -1,31 +1,90 @@
...
+	log.Debugf("%s=%s, resolved to %+v", setting.EnvVar(), val, addr)
 	return nil
 }
 
-// MetricsEnabled returns true if the metrics/debug http server should be started
+func validateTLS() error {
+	certFile := filepath.Join(SecureMetricsCertDir.Setting(), TLSCertFileName)
+	if ok, err := fileutils.Exists(certFile); !ok {
+		if err != nil {
+			log.Warnf("failed to validate secure metrics certificate file %q", certFile)
```
Comment (dhaus67) [inline pkg/env/metrics.go:48]: Why the extra log statement, AFAICT the result of `validateTLS` will be logged anyways?

File: pkg/env/metrics.go [hunk 2]
```diff
@@ -1,31 +1,90 @@
...
+		}
+		return errors.Errorf("secure metrics certificate file %q not found", certFile)
```
Comment (dhaus67) [inline pkg/env/metrics.go:50]: nit: could use `errox.NotFound` here.

File: pkg/env/metrics.go [hunk 3]
```diff
@@ -1,31 +1,90 @@
...
+	}
+
+	keyFile := filepath.Join(SecureMetricsCertDir.Setting(), TLSKeyFileName)
+	if ok, err := fileutils.Exists(keyFile); !ok {
+		if err != nil {
+			log.Warnf("failed to validate secure metrics key file %q", keyFile)
+		}
+		return errors.Errorf("secure metrics key file %q not found", keyFile)
```
Comment (dhaus67) [inline pkg/env/metrics.go:58]: nit: could use errox.NotFound here.

File: pkg/env/metrics.go [hunk 4]
```diff
@@ -1,31 +1,90 @@
...
+		return nil
+	}
+	if err := validateTLS(); err != nil {
+		return err
+	}
+	return validatePort(SecureMetricsPort)
+}
+
+// SecureMetricsEnabled returns true if the metrics/debug https server should be started.
+func SecureMetricsEnabled() bool {
```
Comment (dhaus67) [inline pkg/env/metrics.go:88]: It's a bit unfortunate that we have this conditionally being a value set to `ROX_SECURE_METRICS_PORT`, other environment variables specifically add another boolean environment variable to it. I get why that may be the case for the metrics port, but for the newly introduce one, should it be also "disabled" or rather a boolean flag additionally?

File: pkg/metrics/server.go [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+		}
+	}
+
+	uptimeMetric := prometheus.NewGauge(prometheus.GaugeOpts{
+		Namespace: PrometheusNamespace,
+		Subsystem: subsystem.String(),
+		Name:      "uptime_seconds",
+		Help:      "Total number of seconds that the service has been up",
+	})
+	// Allow the metric to be registered multiple times for tests
```
Comment (dhaus67) [inline pkg/metrics/server.go:57]: nit:
```suggestion
	// Allow the metric to be registered multiple times for tests.
```

File: pkg/env/metrics.go (hunk 1 above)
Comment (stehessel) [reply pkg/env/metrics.go:48]: Ah the point was to also log `err`, which is only returned when there was a problem with stat'ing the file. Whereas a new error is return by `validateTLS` when no file was found, but the check itself was successful but simply returned "file does not exist".

**Review by dhaus67 — APPROVED** [review]

**Pull request merged**